### PR TITLE
Add scheduled_actions variables (#174)

### LINKED ISSUE #174: Allow configuring of Scheduled Actions
Hello guys and thank you for your work on this module.

## Describe the Feature

I would like to have the possibility to configure multiple Scheduled actions in my environment using **additional_settings** or with a dedicated variable.

## Use Case

I'm trying to configure my Beanstalk environment with some scheduled actions to handle my business daily load needs and ensure respect of some security compliance requirements.

Since we can do this using .ebextensions file we could use this method, but we want to be sure on the infrastructure team side that we can apply a mandatory Schedule action that could not be changed by application developments. (.ebextension being specific to this second)

## Describe Ideal Solution

I have 2 solutions at the moment.

**Solution 1**:  We could create a dedicated variable scheduled_actions as following:
```hcl
variable "scheduled_actions" {
  type = list(object({
    name = string
    minsize = string
    maxsize = string
    desiredcapacity = string 
    starttime = string
    endtime = string
    recurrence = string
    suspend = string
  }))
  default     = []
  description = "Define a list of scheduled actions"
}
```

**Advantage**: 
- This variable allow us to declare as many scheduled actions as needed using a dedicated object for each one.
- Easier to specify than declaring each setting (see second idea)

**Drawback**: 

- Some parameters in it are optional but will have to be filled anyway using a null value since we cannot yet specify default values in objects (see: https://github.com/hashicorp/terraform/issues/19898) 
- Add another variable (not sure if it is a drawback here )

**Solution 2**: We could modify the additional_settings variable to handle the "resource" field for this specific case.

https://github.com/cloudposse/terraform-aws-elastic-beanstalk-environment/blob/2d146afbe6ca491d07b81732209244867bbdcd5c/variables.tf#L380-L385

become 
```hcl
variable "additional_settings" {
  type = list(object({
    namespace = string
    name      = string
    value     = string
    resource  = string
  }))
```

https://github.com/cloudposse/terraform-aws-elastic-beanstalk-environment/blob/2d146afbe6ca491d07b81732209244867bbdcd5c/main.tf#L867-L875 become 
```tf
  dynamic "setting" {
    for_each = var.additional_settings
    content {
      namespace = setting.value.namespace
      name      = setting.value.name
      value     = setting.value.value
      resource  = setting.value.resource
    }
  }
```

Advantages: 
- No new variables
- Allow us to declare as many scheduled actions as needed 
- Allow users to handle new settings that would require the specific setting *resource* field later

Drawbacks:
- Sounds like a breaking change since we will change the object variable
- Maybe too heavy in terms of readiness ?

## Alternatives Considered

I'v tried  to use additional_settings without parameter just to see what happen:

I specify additional_settings as following
```hcl
additional_settings = [
  {
    namespace   = "aws:autoscaling:scheduledaction"
    name        = "MinSize"
    value       = "1"
  },
  {
    namespace   = "aws:autoscaling:scheduledaction"
    name        = "MaxSize"
    value       = "2"
  },
  {
    namespace   = "aws:autoscaling:scheduledaction"
    name        = "StartTime"
    value       = "2015-05-14T07:00:00Z"
  },
  {
    namespace   = "aws:autoscaling:scheduledaction"
    name        = "EndTime"
    value       = "2016-01-12T07:00:00Z"
  },
  {
    namespace   = "aws:autoscaling:scheduledaction"
    name        = "Recurrence"
    value       = "*/20 * * * *"
  },
  {
    namespace   = "aws:autoscaling:scheduledaction"
    name        = "DesiredCapacity"
    value       = "2"
  }
]
```

Launch plan wich mention these new settings:
```bash
      + setting {
          + name      = "DesiredCapacity"
          + namespace = "aws:autoscaling:scheduledaction"
          + value     = "2"
        }
      + setting {
          + name      = "EndTime"
          + namespace = "aws:autoscaling:scheduledaction"
          + value     = "2016-01-12T07:00:00Z"
        }
      + setting {
          + name      = "MaxSize"
          + namespace = "aws:autoscaling:scheduledaction"
          + value     = "2"
        }
      + setting {
          + name      = "MinSize"
          + namespace = "aws:autoscaling:scheduledaction"
          + value     = "1"
        }
      + setting {
          + name      = "Recurrence"
          + namespace = "aws:autoscaling:scheduledaction"
          + value     = "*/20 * * * *"
        }
      + setting {
          + name      = "StartTime"
          + namespace = "aws:autoscaling:scheduledaction"
          + value     = "2015-05-14T07:00:00Z"
        }
```
 then apply and get the following error

```bash
Error: InvalidParameterValue: The scheduled action name cannot be blank.
        status code: 400, request id: f4003748-9393-4202-824c-312fbf77b7fc
```

Which is logic since the resource field is set to an empty string.

## Additional Context

This need sounds like a bug and a feature request to me. 

I'm open to discussion on this one and can propose a PR .

### PR DESCRIPTION
**Pull request by florian0410**
## What is changing ?
Adding a new var **scheduled_actions** to give ability to setup beanstalk Scheduled actions.

Example:
```hcl
scheduled_actions = [
  {
    name            = "Refreshinstances"
    minsize         = "1"
    maxsize         = "2"
    desiredcapacity = "2"
    starttime       = "2015-05-14T07:00:00Z"
    endtime         = "2016-01-12T07:00:00Z"
    recurrence      = "*/20 * * * *"
    suspend         = false
  },
  {
    name            = "Refreshinstances2"
    minsize         = "1"
    maxsize         = "2"
    desiredcapacity = "2"
    starttime       = "2015-05-14T07:00:00Z"
    endtime         = "2016-01-12T07:00:00Z"
    recurrence      = "*/30 * * * *"
    suspend         = false
  }
]
```

## Why ?
We cannot use **additional_settings** to define these since it does not handle the dedicated resource field.

https://github.com/cloudposse/terraform-aws-elastic-beanstalk-environment/blob/2d146afbe6ca491d07b81732209244867bbdcd5c/variables.tf#L380-L385

https://github.com/cloudposse/terraform-aws-elastic-beanstalk-environment/blob/2d146afbe6ca491d07b81732209244867bbdcd5c/main.tf#L867-L875

## references
* About resource setting field: https://registry.terraform.io/providers/hashicorp/aws/latest/docs/resources/elastic_beanstalk_environment#resource
* closes #174

### CODE REVIEW

**Comment by florian0410** [comment]
Just updated the code to fix the issue, if you can launch again tests when you can @Gowiem.

Thank you

File: main.tf [hunk 1]
```diff
@@ -811,6 +811,73 @@ resource "aws_elastic_beanstalk_environment" "default" {
...
+    for_each = var.scheduled_actions
+    content {
+      namespace = "aws:autoscaling:scheduledaction"
+      name      = "StartTime"
+      value     = setting.value.starttime
+      resource  = setting.value.name
+    }
+  }
+  dynamic "setting" {
+    for_each = var.scheduled_actions == null ? [] : var.scheduled_actions
```
Comment (Gowiem) [inline main.tf:862]: Doesn't look like you need this logic anymore? Want to update so it's consistent with the rest?

**Review by Gowiem — COMMENTED** [review]
One request for uniformity, but otherwise looks good. 

I'm not 100% sure on the functionality you're adding, but most of us don't work with beanstalk that often anymore so that isn't uncommon. As long as it passed tests and the example works then I'm going to assume this is solid 👍

**Review by joe-niland — APPROVED** [review]

File: main.tf (hunk 1 above)
Comment (florian0410) [reply main.tf:862]: Yeah I'll update it before resolving the conflict

**Review by Gowiem — APPROVED** [review]

**Comment by Gowiem** [comment]
Thanks for the contribution @florian0410! Released as `0.39.0`.

**Pull request merged**